File: julia/collatz.jl
function collatz(seed)
    steps = 0
    while seed > 1
        while seed % 2 == 0
            steps += 1
            seed ÷= 2
        end
        if seed > 1
            steps += 1
            seed = seed * 3 + 1
        end
    end
    return steps
end

function main()
    collatz_result = 0
    for seed in 1:1_000_000
        collatz_result += collatz(seed)
    end
    println("steps $collatz_result")
end

main()

File: perl/collatz.pl
use strict;
use warnings;

sub collatz {
    my $seed = shift;
    my $steps = 0;
    while ($seed > 1) {
        if ($seed % 2 == 0) {
            $seed /= 2;
        } else {
            $seed = $seed * 3 + 1;
        }
        $steps++;
    }
    return $steps;
}

sub main {
    my $collatz_result = 0;
    for my $seed (1 .. 1_000_000) {
        $collatz_result += collatz($seed);
    }
    print "steps $collatz_result\n";
}

main();
